src/huggingface_hub/utils/_deprecation.py

```diff
@@ -0,0 +1,52 @@
+import warnings
+from functools import wraps
+from inspect import Parameter, signature
+
+
+def _deprecate_positional_args(func=None, *, version="0.7"):
+    """Decorator for methods that issues warnings for positional arguments.
+    Using the keyword-only argument syntax in pep 3102, arguments after the
+    * will issue a warning when passed as a positional argument.
+    Parameters
+    ----------
+    func : callable, default=None
+        Function to check arguments on.
+    version : callable, default="0.7"
+        The version when positional arguments will result in error.
+    """
+
+    def _inner_deprecate_positional_args(f):
+        sig = signature(f)
+        kwonly_args = []
+        all_args = []
+        for name, param in sig.parameters.items():
+            if param.kind == Parameter.POSITIONAL_OR_KEYWORD:
+                all_args.append(name)
+            elif param.kind == Parameter.KEYWORD_ONLY:
+                kwonly_args.append(name)
+
+        @wraps(f)
+        def inner_f(*args, **kwargs):
+            extra_args = len(args) - len(all_args)
+            if extra_args <= 0:
+                return f(*args, **kwargs)
+            # extra_args > 0
+            args_msg = [
+                "{}={}".format(name, arg)
```

Comment:
Nitpick, we prefer to use f-strings in general as we find these easier to read

Reply:
hehe, I do too, this is from the time where we still supported py=3.5.